src/Extensions/MemberExtension.php

```diff
@@ -36,58 +36,52 @@ public function onBeforeWrite()
         // Only regenerate if there is no secret and MFA is not enabled yet
         // Inherits MFAEnabled from Bootstrap object extension
         if (!$this->owner->TOTPSecret || !$this->owner->MFAEnabled) {
-            $secret = Totp::GenerateSecret(16);
-            $secret = Base32::encode($secret);
+            $secret = Base32::encodeUpper(random_bytes(128)); // We generate our own 1024 bits secret
             $this->owner->TOTPSecret = $secret;
         }
     }
 
     /**
      * @param FieldList $fields
-     * @throws InvalidWriterException
      */
     public function updateCMSFields(FieldList $fields)
     {
-        if (strlen($this->owner->TOTPSecret)) {
-            $qrcodeURI = $this->GoogleAuthenticatorQRCode();
+        if ($this->owner->TOTPSecret !== '') {
+            $qrcodeURI = $this->getQRCode();
             $fields->addFieldToTab('Root.Main', ToggleCompositeField::create(
                 null,
                 'Second Factor Token Secret',
-                LiteralField::create(null, sprintf("<img src=\"%s\" />", $qrcodeURI))
+                LiteralField::create(null, sprintf('<img src="%s" />', $qrcodeURI))
             ));
             $fields->removeByName('TOTPSecret');
         }
     }
 
     /**
      * @return string
-     * @throws InvalidWriterException
      */
-    public function GoogleAuthenticatorQRCode()
+    protected function getQRCode()
     {
         $qrCode = new QrCode($this->generateOTPAuthString());
         $qrCode->setSize(300);
```

Comment:
Side note: do we want to make the size configurable?

Reply:
It's the size of the image. I think making it configurable is adding quite some overhead for displaying a simple image in the CMS